File: frontend/lib/features/auth/models/register_doctor_request.dart
class RegisterDoctorRequest {
  final String email;
  final String name;
  final String surname;

  RegisterDoctorRequest({
    required this.email,
    required this.name,
    required this.surname,
  });

  Map<String, dynamic> toJson() => {
    'email': email,
    'name': name,
    'surname': surname,
  };
}

File: backend/backend-dialysis-record/src/main/java/com/agustin/backend_dialysis_record/dto/auth/RegisterDoctorRequest.java
package com.agustin.backend_dialysis_record.dto.auth;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record RegisterDoctorRequest(
        @NotBlank
        @Email
        @Size(max = 254)
        String email,

        // password eliminado: la identidad es gestionada por Supabase Auth

        @NotBlank
        @Size(max = 60)
        String name,

        @NotBlank
        @Size(max = 60)
        String surname
) {}